test/upgrades/RibbonThetaYearnVaultWithSwap.ts

```diff
@@ -0,0 +1,170 @@
+import { assert } from "chai";
+import { ethers, network } from "hardhat";
+import {
+  GAMMA_CONTROLLER,
+  MARGIN_POOL,
+  OTOKEN_FACTORY,
+  USDC_ADDRESS,
+  WETH_ADDRESS,
+} from "../../constants/constants";
+import { objectEquals, parseLog, serializeMap } from "../helpers/utils";
+import deployments from "../../constants/deployments.json";
+import { BigNumberish, Contract } from "ethers";
+import * as time from "../helpers/time";
+
+const { parseEther } = ethers.utils;
+
+const UPGRADE_ADMIN = "0x223d59FA315D7693dF4238d1a5748c964E615923";
+const IMPLEMENTATION_SLOT =
+  "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc";
+
+// UPDATE THESE VALUES BEFORE WE ATTEMPT AN UPGRADE
+const FORK_BLOCK = 15177627;
+
+const CHAINID = process.env.CHAINID ? Number(process.env.CHAINID) : 1;
+
+describe("RibbonThetaYearnVault upgrade", () => {
+  let vaults: string[] = [];
+
+  before(async function () {
+    // We need to checkpoint the contract on mainnet to a past block before the upgrade happens
+    // This means the `implementation` is pointing to an old contract
+    await network.provider.request({
+      method: "hardhat_reset",
+      params: [
+        {
+          forking: {
+            jsonRpcUrl: process.env.TEST_URI,
+            blockNumber: FORK_BLOCK,
+          },
+        },
+      ],
+    });
+
+    // Fund & impersonate the admin account
+    const [userSigner] = await ethers.getSigners();
+
+    await userSigner.sendTransaction({
+      to: UPGRADE_ADMIN,
+      value: parseEther("10"),
+    });
+
+    await network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [UPGRADE_ADMIN],
+    });
+
+    const deploymentNames = ["RibbonThetaVaultETHPutYearn"];
+    deploymentNames.forEach((name) => vaults.push(deployments.mainnet[name]));
+  });
+
+  checkIfStorageNotCorrupted(deployments.mainnet.RibbonThetaVaultETHPutYearn);
+});
+
+function checkIfStorageNotCorrupted(vaultAddress: string) {
+  const getVaultStorage = async (storageIndex: BigNumberish) => {
+    return await ethers.provider.getStorageAt(vaultAddress, storageIndex);
+  };
+
+  const variableNames = [
+    "vaultParams",
+    "vaultState",
+    "optionState",
+    "feeRecipient",
+    "keeper",
+    "performanceFee",
+    "managementFee",
+    "optionsPremiumPricer",
+    "strikeSelection",
+    "premiumDiscount",
+    "currentOtokenPremium",
+    "lastStrikeOverrideRound",
+    "overriddenStrikePrice",
+    "auctionDuration",
+    "optionAuctionID",
+    "lastQueuedWithdrawAmount",
```

Comment:
Let's add the variables from RibbonThetaYearnVaultStorageV3 to RibbonThetaYearnVaultStorageV7 to double check

Reply:
done, test passed